protocol/messenger_communities.go

```diff
@@ -3138,6 +3138,17 @@ func (m *Messenger) SlowdownArchivesImport() {
 }
 
 func (m *Messenger) importHistoryArchives(communityID types.HexBytes, cancel chan struct{}) error {
+	community, err := m.communitiesManager.GetByID(communityID)
+	if err != nil {
+		m.communitiesManager.LogStdout("couldn't get community to import archives", zap.Error(err))
+		return err
+	}
+
+	if !community.HasMember(&m.identity.PublicKey) {
+		m.communitiesManager.LogStdout("can't import archives when user not a member of community")
+		return ErrUserNotMember
+	}
```

Comment:
Again, I think if we know that we're not a member we shouldn't be caling `importHistoryArchives` in the first place.
Returning here early is fine, but we should also check the places when membership status changes.

If the client receives a signal that we're no longer a member then we shouldn't be calling `importHistoryArchives` in the first place. At the very least we should check if any import tasks are running and cancel/stop them in case we're no longer a member.

Reply:
Oh sorry, I didn't get it correctly (thought I should do it before starting the loop. Okay I will check

Reply:
Returning early here is perfectly fine I think.
Then we don't have the condition everywhere where we call this function (there might only be one place, I don't actually know from the top of my head).

But we should for sure double check if we're correctly stopping this task in case a membership state changes.